### PR TITLE
Adds trash tag to more items. New component "TrashOnEmpty"

### PR DESCRIPTION
**Pull request by jessicamaybe**
## About the PR 
- Adds trash tag to corn cobs and empty tins

- Adds new component: TrashOnEmpty. It adds the trash tag to items once they've run out of solution. Finally, janitors can throw away the teacups and ketchup packets. 

- Adds the component to: syringes that come with stuff in them, condiment packets, condiment bottles, teacups, and coffee cups

part of #9772

**Changelog**

:cl:
- tweak: Janitors can now throw away more things, including empty teacups and empty ketchup packets!

### CODE REVIEW

File: Content.Server/Nutrition/Components/TrashOnEmptyComponent.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace Content.Server.Nutrition.Components
+{
+
+    [RegisterComponent]
+    public sealed class TrashOnEmptyComponent : Component
+    {
+
+    }
+}
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/Components/TrashOnEmptyComponent.cs:9]: Comments

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Server/Nutrition/EntitySystems/TrashOnEmptySystem.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            if (!EntityManager.HasComponent<SolutionContainerManagerComponent>((component).Owner))
+                return;
+            if (_tagSystem.HasTag(component.Owner, "Trash"))
+                return;
+            if (!_solutionContainerSystem.TryGetDrainableSolution(component.Owner, out var drinkSolution) || drinkSolution.DrainAvailable <= 0)
+            {
+                EntityManager.EnsureComponent<TagComponent>(component.Owner);
+                _tagSystem.AddTag(component.Owner, "Trash");
+            }
+        }
```
Comment (mirrorcult) [inline Content.Server/Nutrition/EntitySystems/TrashOnEmptySystem.cs:29]: this isn't quite robust enough, you'll also want to remove the trash tag if it goes above 0 solution again

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Nutrition/EntitySystems/TrashOnEmptySystem.cs (hunk 2 above)
Comment (mirrorcult) [reply Content.Server/Nutrition/EntitySystems/TrashOnEmptySystem.cs:29]: also maybe want to check on component startup as well?

File: Resources/Prototypes/Entities/Objects/Specific/Medical/healing.yml [hunk 3]
```diff
@@ -232,6 +232,7 @@
         reagents:
         - ReagentId: Ephedrine
           Quantity: 15
+  - type: TrashOnEmpty
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Specific/Medical/healing.yml:235]: with the changes above these can be removed and put on the base `Syringe` instead

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/condiments.yml [hunk 4]
```diff
@@ -357,6 +358,7 @@
     - type: SolutionContainerVisualizer
       maxFillLevels: 6
       fillBaseName: bottle-alpha-
+  - type: TrashOnEmpty
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/condiments.yml:361]: same here, just on the base

Comment (jessicamaybe) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/condiments.yml:362]: the condiment ones already are the base objects, there's just a few different base condiment bottles

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**